### PR TITLE
BS-13861 Add findByPersistenceId provided query

### PR DESCRIPTION
**Pull request by rbioteau**
Add findByPersistenceId provided query.
Do not add the query if already defined in custom queries (avoids
migration issues)

### CODE REVIEW

File: bpm/bonita-util/bonita-common-util/src/main/java/org/bonitasoft/engine/bdm/BDMQueryUtil.java [hunk 1]
```diff
@@ -58,22 +58,22 @@ public static String createQueryNameForUniqueConstraint(final UniqueConstraint u
         if (uniqueConstraint == null) {
             throw new IllegalArgumentException("uniqueConstraint cannot be null");
         }
-        return getQueryName(uniqueConstraint.getFieldNames());
+        return getQueryName(uniqueConstraint.getFieldNames().toArray(new String[0]));
```
Comment (educhastenier) [inline bpm/bonita-util/bonita-common-util/src/main/java/org/bonitasoft/engine/bdm/BDMQueryUtil.java]: What's the interest of changing implem from List to Array?

File: bpm/bonita-util/bonita-common-util/src/main/java/org/bonitasoft/engine/bdm/BDMQueryUtil.java [hunk 2]
```diff
@@ -168,6 +181,13 @@ public static String createQueryContentForField(final String businessObjectName,
     public static List<Query> createProvidedQueriesForBusinessObject(final BusinessObject businessObject) {
         final List<Query> queries = new ArrayList<Query>();
         final Set<String> queryNames = new HashSet<String>();
+
+        if (!containsQuertWithName(businessObject, getQueryName(Field.PERSISTENCE_ID))) {
```
Comment (educhastenier) [inline bpm/bonita-util/bonita-common-util/src/main/java/org/bonitasoft/engine/bdm/BDMQueryUtil.java]: typo "quert" => query

File: bpm/bonita-util/bonita-common-util/src/main/java/org/bonitasoft/engine/bdm/BDMQueryUtil.java (hunk 1 above)
Comment (rbioteau) [inline bpm/bonita-util/bonita-common-util/src/main/java/org/bonitasoft/engine/bdm/BDMQueryUtil.java]: The ... notation is more convenient as we often use the method with only one element.

File: bpm/bonita-util/bonita-common-util/src/main/java/org/bonitasoft/engine/bdm/BDMQueryUtil.java [hunk 3]
```diff
@@ -232,6 +252,14 @@ public static String createQueryNameForLazyField(final BusinessObject businessOb
 
     public static Set<String> getAllProvidedQueriesNameForBusinessObject(final BusinessObject businessObject) {
```
Comment (educhastenier) [inline bpm/bonita-util/bonita-common-util/src/main/java/org/bonitasoft/engine/bdm/BDMQueryUtil.java]: Could we remove "all" in this method name? It does not bring anything

Comment (rbioteau) [inline bpm/bonita-util/bonita-common-util/src/main/java/org/bonitasoft/engine/bdm/BDMQueryUtil.java]: It's API so I don't known.

**Pull request merged**